Src/Newtonsoft.Json/Utilities/LinqBridge.cs

```diff
@@ -72,7 +72,14 @@ public static IEnumerable<TResult> Empty<TResult>()
     {
       CheckNotNull(source, "source");
 
-      return CastYield<TResult>(source);
+        var servesItself = source as IEnumerable<TResult>;
+        if (servesItself != null
+            && (!(servesItself is TResult[]) || servesItself.GetType().GetElementType() == typeof(TResult)))
```

Comment:
This test to avoid array variance causing an issue isn't in the framework linq, which is an in-progress bug. Guarding against it here just in case some 2.0 code in some way hits it and would hence be broken by this change.